webcomponents/core/src/components/reveal/deckdeckgo-reveal/deckdeckgo-reveal.tsx

```diff
@@ -1,4 +1,19 @@
-import {Component, h, Element, Prop, Method, State, Host} from '@stencil/core';
+import { Component, h, Element, Prop, Method, State, Host, FunctionalComponent } from '@stencil/core';
+
+const arrayOf = (len: number) => new Array(len).fill(0).map((_, i) => i);
+
+const RevealNthChild: FunctionalComponent<{ index: number }> = ({ index }) => {
```

Comment:
FunctionalComponent which will render a `<style />` tag targeting the given `nth-child`.

> I also tried using a single style tag and dynamically adding the `::slotted(...)` declarations to it as needed, but the browser didn't seem to register the styles when mutated.

This could be moved to the bottom of the file for readability?